### PR TITLE
Use UnixSocket instead of SocketManager in proxy

### PR DESCRIPTION
**Pull request by JosiahWI**
We had hoped it would be simple to replace the use of `SocketManager` in all
components in one go. Sadly, the use of `SocketManager` is inconsistent and
mixed with direct calls to functions from ink_sock.h. Since there will be
changes to the UnixSocket API to fix the consistency, I am making these
changes one component at a time for now.

### CODE REVIEW

File: src/proxy/http/HttpConfig.cc [hunk 1]
```diff
@@ -32,6 +32,7 @@
 #include "proxy/http/HttpConfig.h"
 #include "proxy/hdrs/HTTP.h"
 #include "iocore/eventsystem/ConfigProcessor.h"
+#include "iocore/eventsystem/UnixSocket.h"
```
Comment (JosiahWI) [inline src/proxy/http/HttpConfig.cc:35]: The iocore/eventsystem/SocketManager.h file was not included as it should have been; I did not forget to remove it here.

**Comment by JosiahWI** [comment]
```
  45/134 Test  #40: test_jsonrpcserver .....................***Failed    3.86 sec
[Jul 11 06:00:23.431]  DIAG: <IPCSocketServer.cc:161 (init)> (rpc.net) Using tests/var/jsonrpc20_test.sock as socket path.
[Jul 11 06:00:23.431]  DIAG: <RPCServer.cc:74 (start_thread)> (rpc) Starting RPC Server on: Local Socket
[Jul 11 06:00:23.433]  DIAG: <JsonRPCManager.cc:211 (handle_call)> (rpc.msg) --> JSONRPC request
'{"jsonrpc": "2.0", "method": "some_foo", "params": {"duration": 1}, "id": "aBcD"}'
[Jul 11 06:00:24.437]  DIAG: <JsonRPCManager.cc:256 (handle_call)> (rpc.msg) <-- JSONRPC Response
 '{"jsonrpc": "2.0", "result": {"res": "ok", "duration": "1"}, "id": "aBcD"}'
[Jul 11 06:00:24.438]  DIAG: <JsonRPCManager.cc:211 (handle_call)> (rpc.msg) --> JSONRPC request
'{"jsonrpc": "2.0", "method": "some_foo", "params": {"duration": 1}, "id": "eFgH"}'
[Jul 11 06:00:25.441]  DIAG: <JsonRPCManager.cc:256 (handle_call)> (rpc.msg) <-- JSONRPC Response
 '{"jsonrpc": "2.0", "result": {"res": "ok", "duration": "1"}, "id": "eFgH"}'
[Jul 11 06:00:25.443]  DIAG: <JsonRPCManager.cc:211 (handle_call)> (rpc.msg) --> JSONRPC request
'{"jsonrpc": "2.0", "method": "do_nothing", "params": {"msg":"5PN2qmWqBlQ9wQj99nsQzldVI5ZuGXbEWRK5RhRXdCdG5nG5azdNMK66MuCV6GXi5xr84P2R391UXaLHbavJvFZGfO47XWS2qVOw5lxuBEBNue7b9PZS0hoI6pgabi9UOuSpM2Tlh01vOzwvSikHFswuzleht6xGbotuQQgmO6GkdAqwedXNRRadbLMZROaAfZDAWhxgqRqcM1fdhRxRZYw3GFhPr6BInZITSBQOgwxhHxrhytFUloiivhOmebpblnTipx7wXZAqJZR5Y4M9k8AIyGE9CpuXkn1XvnFttynI1MguokYDeWl5noesnB4mh19PM6Dph2Nh722AggBfUKJEdmm09Z1uFII2PNfTdMjJFB59lvpzORCPO4yy5dZ4yR2nIGRwH4cYCroORc958dBikjYDvOcFEzUEcMGNuUgtShrdNIwenrKHe3wXIoqLf5wyihrXUKPqBV6CnbV3qbB580J4Frwx5ufHb1WO8AK6OoG9srCT"}, "id":"EfGh-1"}'
[Jul 11 06:00:25.444]  DIAG: <JsonRPCManager.cc:256 (handle_call)> (rpc.msg) <-- JSONRPC Response
 '{"jsonrpc": "2.0", "result": {"size": "500"}, "id": "EfGh-1"}'
[Jul 11 06:00:25.444]  DIAG: <IPCSocketServer.cc:238 (run)> (rpc.net) Error detected while reading: Peer disconnected after reading 24576 bytes.
```

**Comment by JosiahWI** [comment]
[approve ci osx]

**Comment by JosiahWI** [comment]
[approve ci osx]

**Review by maskit — APPROVED** [review]

**Pull request merged**